pallets/interest-accrual/src/lib.rs

```diff
@@ -383,6 +421,23 @@ pub mod pallet {
 			});
 			T::DbWeight::get().reads_writes(count, count)
 		}
+
+		pub fn count_rates() -> Weight {
+			let count = Rate::<T>::iter_keys().count();
+			RateCount::<T>::set(count as u32);
+			T::DbWeight::get().reads(count as u64) + T::DbWeight::get().writes(1)
+		}
+
+		pub(crate) fn validate_rate(interest_rate_per_year: T::InterestRate) -> DispatchResult {
+			let four_decimals = T::InterestRate::saturating_from_integer(10000);
+			ensure!(
+				interest_rate_per_year < One::one()
+					&& interest_rate_per_year > Zero::zero()
+					&& (interest_rate_per_year * four_decimals).frac() == Zero::zero(),
```

Comment:
`saturating_mul` although I know the booleans will fail before reaching this case.